### PR TITLE
tidb_exporter: add multiple tidb servers support

### PR DESCRIPTION
**Pull request by LinuxGit**
tidb_exporter:
- add multiple tidb servers support
- modify default listen port from 9200 to 9500 to avoid collision with Elasticsearch
- don't check `--tidb.password` parameter, and allow empty password

### CODE REVIEW

File: tidb_exporter/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 TiDB exporter
 ------
 
-TiDB exporter for Prometheus. The `tidb_exporter` will expose metrics via HTTP on the `/metrics` endpoint. Currently only one tidb server address is supported. It is recommended to set load balancing IP. We'll support multiple tidb servers in the near future.
+TiDB exporter for Prometheus. Checking tidb server's health by sending SQL query to it. The `tidb_exporter` will expose metrics via HTTP on the `/metrics` endpoint.
```
Comment (GregoryIan) [inline tidb_exporter/README.md]: @lilin90 PTAL readme

**Comment by WangXiangUSTC** [comment]
LGTM

File: tidb_exporter/main.go [hunk 2]
```diff
@@ -17,27 +19,37 @@ import (
...
+type tidbNode struct {
+	addr string
+	db   *sql.DB
 }
 
 type tidbOpts struct {
-	address  string
-	username string
-	password string
+	addresses string
```
Comment (GregoryIan) [inline tidb_exporter/main.go]: addrs

File: tidb_exporter/main.go [hunk 3]
```diff
@@ -17,27 +19,37 @@ import (
...
+	username  string
+	password  string
 }
 
 // NewExporter returns an initialized Exporter.
 func NewExporter(opts tidbOpts) (*Exporter, error) {
-	db, err := accessDatabase(opts.username, opts.password, opts.address, dbname)
-	if err != nil {
-		return nil, errors.Trace(err)
+	addresses := strings.Split(opts.addresses, ",")
```
Comment (GregoryIan) [inline tidb_exporter/main.go]: trim and verify them

File: tidb_exporter/README.md (hunk 1 above)
Comment (lilin90) [inline tidb_exporter/README.md]: Change this part to:

TiDB exporter (for Prometheus) is a tool for checking the TiDB server's health by sending SQL queries to it. The `tidb_exporter` exposes metrics via HTTP on the `/metrics` endpoint.

File: tidb_exporter/main.go [hunk 4]
```diff
@@ -17,27 +20,60 @@ import (
...
 	if err != nil {
-		return nil, errors.Trace(err)
+		return nil, err
 	}
 
-	return &Exporter{
-		db:       db,
-		tidbOpts: opts,
-	}, nil
+	nodes := []tidbNode{}
```
Comment (GregoryIan) [inline tidb_exporter/main.go]:
```
nodes := make([]tidbNode, 0, len(addrs))
```

**Pull request merged**